src/types.rs

```diff
@@ -921,3 +921,87 @@ impl Debug for Log {
             .finish()
     }
 }
+
+/// An Ethereum Transaction Receipt
+#[derive(Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionReceipt {
+    #[serde(rename = "type", flatten)]
+    pub kind: TransactionReceiptKind,
+
+    /// The hash of the transaction that emitted this log.
+    pub transaction_hash: Digest,
+    /// The index of the transaction that emitted this log within the block.
+    pub transaction_index: U256,
+    /// The hash of the block containing the transaction.
+    pub block_hash: Digest,
+    /// The height of the block containing the transaction.
+    pub block_number: U256,
+    /// Address of transaction sender.
+    pub from: Address,
+    /// Transaction receipient (Null for contract creation).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to: Option<Address>,
+    /// The sum of gas used by this transaction and all preceding transactions in the same block.
+    pub cumulative_gas_used: U256,
+    /// The amount of gas used for this specific transaction alone.
+    pub gas_used: U256,
+    /// Contract address created, or `None` if not a deployment.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub contract_address: Option<Address>,
+    /// Logs emitted by the transaction.
+    pub logs: Vec<Log>,
+    /// The log bloom filter.
+    pub logs_bloom: Bloom,
+    /// State root. Only present before activation of [EIP-658](https://eips.ethereum.org/EIPS/eip-658).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub root: Option<Digest>,
+    /// Status: either 1 (success) or 0 (failure). Only present after activation of [EIP-658](https://eips.ethereum.org/EIPS/eip-658).
+    pub status: Option<U256>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum TransactionReceiptKind {
+    #[serde(rename = "0x0")]
+    Legacy,
+    #[serde(rename = "0x1")]
+    Eip2930,
+    #[serde(rename = "0x2")]
+    Eip1559 {
+        /// The price paid post-execution by the transaction (i.e. base fee + priority fee).
+        /// Both fields in 1559-style transactions are *maximums* (max fee + max priority fee), the
+        /// amount that's actually paid by users can only be determined post-execution.
+        effective_gas_price: Option<U256>,
```

Comment:
Why is this an `Option`? And are you sure this is only populated for ERC-1559 transactions?

Reply:
I just checked, and this field is always set, even for transactions that don't have a type of `0x2`. Proof:

```
$ curl -s -X POST -H 'content-type: application/json' https://mainnet.infura.io/v3/$INFURA_PROJECT_ID --data '{"jsonrpc":"2.0","id":0,"method":"eth_getTransactionReceipt","params":["0x699b6e8290c191e639fcec1adb31d81a88b7d2ac54486c333cd603ada3fb1d2d"]}' | jq '.result | { type, effectiveGasPrice }'
{
  "type": "0x0",
  "effectiveGasPrice": "0x5fe3cb8f0"
}
```

Reply:
Moved out of enum and made not to be an option.